### PR TITLE
add `get_logs` testing

### PR DESCRIPTION
**Pull request by tcoratger**
Time spent on this PR:

Resolves: #1055 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: tests/tests/eth_provider.rs [hunk 1]
```diff
@@ -283,14 +283,93 @@ async fn test_get_logs_block_range(#[future] katana: Katana, _setup: ()) {
...
+    );
+
+    // Create a vector to store a few addresses.
+    let mut some_addresses = Vec::new();
+    // Add two addresses from the retrieved logs.
+    for log in &all_logs_katana {
+        if some_addresses.len() < 2 {
+            some_addresses.push(log.address());
+        }
+    }
```
Comment (greged93) [inline tests/tests/eth_provider.rs:356]: could we do `all_logs_katana.clone().into_iter().take(2).collect()`?

File: tests/tests/eth_provider.rs [hunk 2]
```diff
@@ -283,14 +283,93 @@ async fn test_get_logs_block_range(#[future] katana: Katana, _setup: ()) {
...
+        filter_logs(Filter::new().address(some_addresses.clone()), provider.clone()).await,
+        katana.logs_by_address(&some_addresses)
+    );
+
+    // Create a vector to store all addresses.
+    let mut all_addresses = Vec::new();
+    // Add all addresses from the retrieved logs.
+    for log in &all_logs_katana {
+        all_addresses.push(log.address());
+    }
```
Comment (greged93) [inline tests/tests/eth_provider.rs:368]: could we do `all_logs_katana.clone().into_iter().map(|l| l.address()).collect()`?

File: tests/tests/eth_provider.rs (hunk 1 above)
Comment (tcoratger) [reply tests/tests/eth_provider.rs]: yes, fixed

File: tests/tests/eth_provider.rs (hunk 2 above)
Comment (tcoratger) [reply tests/tests/eth_provider.rs]: fixed

**Review by greged93 — APPROVED** [review]

**Pull request merged**